ipaserver/plugins/hbacrule.py

```diff
@@ -425,67 +425,61 @@ def execute(self, cn, **options):
         )
 
 
-# @register()
-class hbacrule_add_accesstime(LDAPQuery):
-    """
-    Add an access time to an HBAC rule.
-    """
+@register()
+class hbacrule_add_timerule(LDAPAddMember):
+    __doc__ = _('Add time rules to an HBAC rule.')
 
-    takes_options = (
-        AccessTime('accesstime',
-            cli_name='time',
-            label=_('Access time'),
-        ),
-    )
+    member_attributes = ['membertimerule']
+    member_count_out = ('%i object added.', '%i objects added.')
 
-    def execute(self, cn, **options):
+    def execute(self, *args, **options):
         ldap = self.obj.backend
+        dn = self.obj.get_dn(*args, **options)
+        assert(isinstance(dn, DN))
 
-        dn = self.obj.get_dn(cn)
-
-        entry_attrs = ldap.get_entry(dn, ['accesstime'])
-        entry_attrs.setdefault('accesstime', []).append(
-            options['accesstime']
-        )
         try:
-            ldap.update_entry(entry_attrs)
-        except errors.EmptyModlist:
-            pass
+            entry_attrs = ldap.get_entry(dn, ['objectclass'])
         except errors.NotFound:
-            self.obj.handle_not_found(cn)
+            self.obj.handle_not_found(*args)
+        objclass_updated = False
+        if ('ipahbacrulev2' not in
+                (o.lower() for o in entry_attrs['objectclass'])):
+            entry_attrs['objectclass'] = ['ipaassociation', 'ipahbacrulev2']
```

Comment:
Be nice for future me or you, and don't hardcore used objectclass. Only remove 'ipahbacrule' and only add 'ipahbacrulev2'. We may add new objectclasses in future and I don't want to change all places :)